subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitTestDryRunner.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.junit;
+
+import org.gradle.api.NonNullApi;
+import org.junit.runner.Description;
+import org.junit.runner.Runner;
+import org.junit.runner.notification.RunNotifier;
+
+import java.util.LinkedList;
+
+@NonNullApi
+public class JUnitTestDryRunner extends Runner {
+    private final Runner runner;
+
+    public JUnitTestDryRunner(Runner runner) {
+        this.runner = runner;
+    }
+
+    @Override
+    public Description getDescription() {
+        return runner.getDescription();
+    }
+
+    @Override
+    public void run(RunNotifier notifier) {
+        LinkedList<Description> queue = new LinkedList<Description>();
+        queue.add(runner.getDescription());
+        while (!queue.isEmpty()) {
+            Description description = queue.removeFirst();
+            queue.addAll(description.getChildren());
+            if (JUnitTestEventAdapter.methodName(description) != null) {
+                notifier.fireTestIgnored(description);
```

Comment:
I'm not entirely sure if this is sufficient in all cases. There might some custom runners that don't adhere to the naming convention. I'm wondering if doing this for all leaf notes (where `description.isTest()` returns `true`) would be less brittle.

Reply:
Changed to `description.isTest()` check, but not sure how to validate it.